.github/workflows/main.yaml

```diff
@@ -62,14 +62,28 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           fail_ci_if_error: false
+  publish:
+    runs-on: ubuntu-latest
+    needs: test
+    if: github.ref == 'refs/heads/master'
+    environment: CD
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - run: |
+          git rev-list --count --first-parent HEAD > patch_version.txt
+      - name: Set up JDK 8
+        uses: coursier/setup-action@v1
+        with:
+          jvm: adopt:8
       - name: Build JAR file
         run: |
           LIB_PACKAGE="+library/package"
```

Comment:
I'd suggest keeping at least `sbt -Dsbt.log.noformat=true +{,macroSub/,structType/}dependencyLockCheck` somewhere in publish flow too. If build-test and publish are separated it means publish (retry) can happen much later and so external artifacts contents might have changed (even better would be if sbt enforced them, not just validated on demand..)

Reply:
Makes sense